src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,57 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_SIZE = 10;
+    private int currentSize = INITIAL_SIZE;
+    private int size = 0;
+    private T[] table = (T[]) new Object[currentSize];
 
+    public void push(T value) {
+        if (size >= currentSize) {
+            table = grow();
+        }
```

Comment:
Consider checking for the array growth condition inside the `grow` method. This encapsulation will make the `push` method cleaner and the `grow` method more self-contained.

Reply:
ok, done